ios/RNTPTDocumentView.m

```diff
@@ -2511,6 +2511,24 @@ - (void)applyDocumentControllerSettings:(PTDocumentController *)documentControll
         }
         documentController.toolbarItems = [bottomToolbarItems copy];
     }
+    
+    // Override action of overridden toolbar button items
+    if (self.overrideToolbarButtonBehavior) {
+        for (NSString *buttonString in self.overrideToolbarButtonBehavior) {
+            UIBarButtonItem *toolbarItem = [self itemForButton:buttonString
+                                                 inViewController:documentController];
+            
+            NSString *actionName = [NSString stringWithFormat:@"overriddenPressed_%@",
+                                    buttonString];
+            const SEL selector = NSSelectorFromString(actionName);
+            
+            RNTPT_addMethod([documentController class], selector, ^(id documentController) {
+                [self.delegate toolbarButtonPressed:self withKey:buttonString];
```

Comment:
This is very unconventional.... where the DocumentController will call the DocumentView's delegate object.

Does it work if you instead add the method to the RNTPTDocumentVIew, which will then call its own delegate?

Reply:
if i add the method to RNTPTDocumentView, pressing the button throws a runtime error saying unrecognized selector

Reply:
An alternative / more conventional way to do this would be to have all buttons would call the same method with themselves as the argument, and then that method would use information on the sender object to determine what to do. Is this possible?

Reply:
if i'm correct the bar button items aren't identifiable by their ID string so i would need to define a helper method to convert between the UIBarButtonItem object and its string but i think it's possible? i'll try it that way

Reply:
You could also extend the class to add whatever info you need, might be easier than writing a translating helper method.